processor/src/memory/mod.rs

```diff
@@ -224,16 +224,45 @@ impl Memory {
         self.get_values(RangeInclusive::new(0, u64::MAX))
     }
 
-    /// Returns values within a range of addresses.
+    /// Returns values within a range of addresses at the last clock cycle.
     pub fn get_values(&self, range: RangeInclusive<u64>) -> Vec<(u64, Word)> {
         let mut data: Vec<(u64, Word)> = Vec::new();
+
         for (&addr, addr_trace) in self.trace.range(range) {
             let value = addr_trace.last().expect("empty address trace").1;
             data.push((addr, value));
         }
 
         data
     }
+
+    /// Returns values within a range of addresses, or optionally all values at a partciular step.
+    /// If no step is specified, then values at the last step are returned.
```

Comment:
This comment needs to be updated (probably worth mentioning that this returns memory state at the beginning of the specified cycle). Also, typo in "partciular".

Reply:
fixed